group_project_v2/mixins.py

```diff
@@ -133,13 +133,28 @@ def _get_setting(self, setting, default):
         return result
 
 
-class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin, CourseAwareXBlockMixin):
 
-    DEFAULT_TA_ROLE = ('assistant',)
+    DEFAULT_TA_ROLE = ("assistant", )
 
     _ACCESS_DASHBOARD_ROLE_GROUPS_KEY = "access_dashboard_groups"
     _ACCESS_DASHBOARD_FOR_ALL_ORGS_GROUPS_KEY = "access_dashboard_for_all_orgs_groups"
-    _TA_ROLES_KEY = 'ta_roles'
+    _ACCESS_DASHBOARD_TA_ROLES = "access_dashboard_ta_groups"
+    _TA_ROLES_KEY = "ta_roles"
+
+    @property
+    def see_dashboard_ta_perms(self):
+        """
+        :return: Returns a list of group names, user needs to be a member of
+                 any group from this list to access dashboard.
+                 Additionally if he is in one of these groups he needs to
+                 have a appropriate role for the course to see the dashboard.
+
+                 Essentially he needs to pass ``check_ta_access`` test.
```

Comment:
@jbzdak nit: slightly misleading docstring - the first sentence reads as if being the member of this role is the only condition to access dashboard. It is clarified by the second sentence, but it would be good to make it clear from the start.